src/codeu/chat/server/BroadCastSystem.java

```diff
@@ -0,0 +1,148 @@
+                while (true) {
+                    broadCastMessage(messagesToBroadcast.take());
+                }
+            } catch (InterruptedException exc) {
+                System.out.println("Error broadcasting message");
+            }
+
+        }
+    }
+
+    public BroadCastSystem(){
+        conversationUsers = new ConcurrentHashMap<>();
+        messagesToBroadcast = new LinkedBlockingQueue<>();
+
+        MessageBroadCaster messageBroadCaster = new MessageBroadCaster();
+
+        broadCaster = new Thread(messageBroadCaster);
+
+        broadCaster.start();
+    }
+
+    private void addConnection(Connection connection, Uuid uuid){
+
+        if (connection == null) throw new NullPointerException();
+        List<Connection> recipients = conversationUsers.get(uuid.id());
+        recipients.add(connection);
+
+    }
+
+    private void removeConnection(Connection connection, Uuid uuid){
+
+        if (connection == null) throw new NullPointerException();
+        List<Connection> recipients = conversationUsers.get(uuid.id());
+        recipients.remove(connection);
+
+    }
+
+    public void switchConversation(Connection connection, ConversationSummary oldCon, ConversationSummary newCon){
+
+        if (newCon == null && oldCon == null)
+            throw new NullPointerException();
+        if (oldCon != null)
+            removeConnection(connection, oldCon.id);
+        if (newCon != null)
+            addConnection(connection, newCon.id);
+
+    }
+
+    private void broadCastMessage(ConversationMessageLink messageLink){
+
+
+        int conversationId = messageLink.conversationUuid.id();
+        Message message = messageLink.message;
+
+        // using an iterator in order to remove connections if they return an exception
+        // this is in case the client has disconnected for whatever reason
+
+        Iterator<Connection> myIterator = conversationUsers.get(conversationId).iterator();
+
+        while (myIterator.hasNext()) {
+            Connection connection = myIterator.next();
+
+            try {
+
+                OutputStream out = connection.out();
+
+                synchronized (out) {
+                    Serializers.INTEGER.write(out, NetworkCode.NEW_BROADCAST);
+                    Message.SERIALIZER.write(out, message);
+                }
+
+            } catch (Exception ex) {
+                try {
+                    connection.close();
+                } catch (IOException ioexc) {
+                    System.out.println("Error closing socket");
+                }
+                myIterator.remove();
+            }
+
+        }
+
+    }
+
+    public void addConversation(ConversationSummary conversationSummary){
+
+        if (conversationSummary == null) throw new NullPointerException("Cannot add null to conversations");
+        if (conversationSummary.id == null) throw new NullPointerException("Null ID in conversation summary");
+
+        int uid = conversationSummary.id.id();
```

Comment:
I took another look at the provided UUID class.  If I'm understanding it correctly, it's just a chain of integers (since typically UUIDs are much bigger than int space).  I think your use of uuid.id() throughout the code is incorrect, since it's effectively just using the last int in the chain as a unique identifier.